driver/linux/main.c

```diff
@@ -299,6 +301,19 @@ static int __init sgx_init(void)
 
 	if (!sgx_page_cache_init())
 		return -EFAULT;
+#ifdef HAVE_MMPUT_ASYNC
+	k_mmput_async = mmput_async;
+#else
+#ifdef HAVE_KSYM_LOOKUP
+	k_mmput_async = (void*)kallsyms_lookup_name("mmput_async");
+#else
+	#error "kernel version is not supported. We need either mmput_async or kallsyms_lookup_name exported from the kernel"
+#endif
+#endif
+	if (!k_mmput_async){
```

Comment:
Is this check defense-in-depth?
We could get an error at compile time if neither feature is present.
When can we get a runtime error? When kallsyms_lookup_name() can't find the mmput_async symbol?

Reply:
yeah, just in case for some kernel,  kallsyms_lookup_name() can't find the mmput_async symbol. I think it can happen in old kernels.